### ISSUE TITLE
Problem when trying to install uplink 0.8.0.post1 

### ISSUE THREAD

**Issue by liiight** [prkumar/uplink#148]
**Describe the bug**
Crash when trying to install uplink 0.8.0.post1

**To Reproduce**
I install via `setup test`. This is my `setup.py`:
```python
from pathlib import Path

from setuptools import setup, find_packages

version = Path('VERSION').read_text()

setup(
    name='shield',
    version=version,
    packages=find_packages(),
    url='https://bitbucket.org/firelayers/shield/',
    license='',
    author='orcarmi',
    author_email='',
    description='Proofpoint python testing',
    setup_requires=['pytest-runner'],
    tests_require=Path('requirements.txt').read_text(),
)
```
This is the full error:
```
Running uplink-0.8.0.post1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-denwjpmw/uplink-0.8.0.post1/egg-dist-tmp-234m0q2w
Traceback (most recent call last):
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 154, in save_modules
    yield saved
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 250, in run_setup
    _execfile(setup_script, ns)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 45, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-denwjpmw/uplink-0.8.0.post1/setup.py", line 37, in <module>
  File "/tmp/easy_install-denwjpmw/uplink-0.8.0.post1/setup.py", line 8, in read
    name='shield',
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 5128: ordinal not in range(128)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 17, in <module>
    tests_require=Path('requirements.txt').read_text(),
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/jenkins/workspace/execute_shield_pull_request_dlp_email_smoke_test_dev/.eggs/pytest_runner-4.4-py3.6.egg/ptr.py", line 183, in run
    installed_dists = self.install_dists(dist)
  File "/home/jenkins/workspace/execute_shield_pull_request_dlp_email_smoke_test_dev/.eggs/pytest_runner-4.4-py3.6.egg/ptr.py", line 131, in install_dists
    orig.test.install_dists(dist),
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/command/test.py", line 208, in install_dists
    tr_d = dist.fetch_build_eggs(dist.tests_require or [])
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/dist.py", line 724, in fetch_build_eggs
    replace_conflicting=True,
  File "/home/jenkins/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 782, in resolve
    replace_conflicting=replace_conflicting
  File "/home/jenkins/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1065, in best_match
    return self.obtain(req, installer)
  File "/home/jenkins/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1077, in obtain
    return installer(requirement)
  File "/home/jenkins/workspace/execute_shield_pull_request_dlp_email_smoke_test_dev/.eggs/pytest_runner-4.4-py3.6.egg/ptr.py", line 72, in fetch_build_egg
    return cmd.easy_install(req)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 679, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 705, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 890, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 1158, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 1144, in run_setup
    run_setup(setup_script, args)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 253, in run_setup
    raise
  File "/usr/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/usr/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 166, in save_modules
    saved_exc.resume()
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 141, in resume
    six.reraise(type, exc, self._tb)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/_vendor/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 154, in save_modules
    yield saved
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 250, in run_setup
    _execfile(setup_script, ns)
  File "/home/jenkins/.local/lib/python3.6/site-packages/setuptools/sandbox.py", line 45, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-denwjpmw/uplink-0.8.0.post1/setup.py", line 37, in <module>
  File "/tmp/easy_install-denwjpmw/uplink-0.8.0.post1/setup.py", line 8, in read
    name='shield',
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 5128: ordinal not in range(128)
```
**Expected behavior**
Should be installed. I'm not sure where that decode issue comes, I imagine it's from the `.post1` version suffix. I tried forcing the file to be ascii encoded but had not luck.

Also note the command that is used to install uplink:
`Running uplink-0.8.0.post1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-denwjpmw/uplink-0.8.0.post1/egg-dist-tmp-234m0q2w`

**Additional context**
This work locally on my mac but fails when running in Jenkins CI.

**Comment by ppslim** [comment]
The `setup.py` provides a `read` method that i non-Locale safe for reading from files.

File 37 reads the `README.rst` which contains several UTF-8 characters, including a `’` at line 150, byte 5128. There are some additional ones further down, including "💜" it was as of yet to fail on.

On systems with Ascii as the default encoding, the read will fail with a trace similar to that in the description. A sample test for this
```
$ virtualenv -p python3 py3
$ . py3/bin/activate
$ git clone https://github.com/prkumar/uplink.git
$ cd uplink
$ git checkout tags/v0.8.0
$ LC_ALL=C python setup.py
Traceback (most recent call last):
  File "setup.py", line 37, in <module>
    "long_description": read("README.rst"),
  File "setup.py", line 8, in read
    return stream.read()
  File "/home/d41d8cd98f00b204e9800998ecf8427e/py3/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 5128: ordinal not in range(128)
```

The readme needs to be sanitised of this content, or incorrect content ignored.
e.g.
```python
def read(filename):
    with open(filename, 'rb') as stream:
        return stream.read().decode('ascii', 'ignore')
```

**Comment by liiight** [comment]
Wow dude, nice find!

**Comment by prkumar** [comment]
@ppslim - Thanks for the great work!! 

As you noted, seems like there are two lines containing non-ASCII characters.
![Image 3-11-19 at 11 48 AM](https://user-images.githubusercontent.com/10181244/54149574-b2f49c80-43f3-11e9-8fac-7056cf0b963a.jpg)

I've committed changes to the [make-README-ascii](https://github.com/prkumar/uplink/tree/make-README-ascii) branch that should fix the issue -- the README is now recognized as an ASCII file:
![Image 3-11-19 at 11 56 AM](https://user-images.githubusercontent.com/10181244/54150002-b2a8d100-43f4-11e9-86f7-e9651f426cba.jpg)

FYI: @liiight

**Comment by prkumar** [comment]
@liiight, @ppslim - We can release this ASAP as a hot-fix. To that end, are there any preferences regarding whether the release should be `0.8.1` or `0.8.0.post2`. The case for incrementing the post version is that the release doesn't include any API or code changes to the main library. However, this is ultimately our decision. If there's no opposition to incrementing the post version, I'm leaning towards that, but I'm open to feedback!

**Comment by liiight** [comment]
i think post2 fits here, as it mostly used for doc changes, which this certainly is.

**Comment by liiight** [comment]
From gitter by @ppslim:

> A quick review of this though shows that on Pypi, v0.7.0 has a wheel, but v0.8.0.a0 and above do not
The content of readme merged into the wheel METADATA at build time and pip prefers wheels
I've not looked deeper, but this seems to imply that even though the wheel still has the UTF-8 characters, it is self regulating
This leads to the following commit
https://github.com/prkumar/uplink/commit/85971f5044dbca358a1b24c9181de56b99403582#diff-354f30a63fb0907d4ad57269548329e3

**Comment by prkumar** [comment]
Fix should be live now! Upgrade to `v0.8.0.post2` when you get the chance 😃

**Comment by liiight** [comment]
Verified. Thanks @prkumar and @ppslim!
